includes/classes/sync/pull.php

```diff
@@ -365,15 +365,35 @@ protected function loop_item_elements_and_map( $post_data ) {
 					$is_component_repeatable = ( is_object( $metadata ) && isset( $metadata->repeatable ) ) ? $metadata->repeatable->isRepeatable : false;
 				}
 				
-				foreach ( $fields_data as $field_data ) {
-					$this->element = (object) $this->format_element_data( $field_data, $component_uuid, true, $is_component_repeatable);
-					$destination   = $this->mapping->data( $this->element->name );
+				$componentProcessed = false; // Initialize flag outside the loop
 
-					if ( $destination && isset( $destination['type'], $destination['value'] ) ) {
-						$columns[ $destination['value'] ] = true;
-						$post_data                        = $this->set_post_values( $destination, $post_data );
+				foreach ($fields_data as $field_data) {
```

Comment:
The old procedure to generate the uuid ($this->element) does not account for components so we need to refactor for that.
This is the identified pattern used for the uuid;
normal fields: [name] = [uuid]
fields in components: [name] = [field uuid]_component_[component uuid]
components: name = [component uuid]_component_[component uuid]